crates/symbolicator-service/src/services/symbolication/symbolication.rs

```diff
@@ -135,31 +136,27 @@ fn symbolicate_frame(
     frame.package = lookup_result.object_info.raw.code_file.clone();
 
     match lookup_result.cache {
-        Some(CacheFile::SymCache(symcache)) => {
-            symbolicate_native_frame(symcache, lookup_result, registers, signal, frame, index)
-        }
-        Some(CacheFile::PortablePdbCache(ppdbcache)) => {
-            symbolicate_dotnet_frame(ppdbcache, frame, index)
-        }
-        None if lookup_result.object_info.debug_status == ObjectFileStatus::Malformed => {
-            Err(FrameStatus::Malformed)
+        CacheEntry::AllGood(CacheFileEntry::SymCache(symcache)) => symbolicate_native_frame(
+            symcache.get(),
+            lookup_result,
+            registers,
+            signal,
+            frame,
+            index,
+        ),
+        CacheEntry::AllGood(CacheFileEntry::PortablePdbCache(ppdb_cache)) => {
+            symbolicate_dotnet_frame(ppdb_cache.get(), frame, index)
         }
-        None => Err(FrameStatus::Missing),
+        CacheEntry::Malformed(_) => Err(FrameStatus::Malformed),
+        _ => Err(FrameStatus::Missing),
```

Comment:
Does `FrameStatus` have more options that we can use now?

Reply:
The definition of `FrameStatus` is
```
pub enum FrameStatus {
    /// The frame was symbolicated successfully.
    Symbolicated,
    /// The symbol (i.e. function) was not found within the debug file.
    MissingSymbol,
    /// No debug image is specified for the address of the frame.
    UnknownImage,
    /// The debug file could not be retrieved from any of the sources.
    Missing,
    /// The retrieved debug file could not be processed.
    Malformed,
}
```
I don't think any of these except for `Missing` and `Malformed` are useful here.